### ISSUE TITLE
Pinpoint doesn't register notification token since 2.6.14 version

### ISSUE THREAD

**Issue by karpov3** [aws/aws-sdk-ios#886]
* What service are you using?
**Pinpoint**

* In what version of SDK are you facing the problem? 
**>=2.6.14**

* Is the issue limited to Simulators / Actual Devices? 
**NO**

* Can your problem be resolved if you bump to a higher version of SDK? 
**NO**

* Is this problem related to specific iOS version? 
**NO**

* How are you consuming the SDK? CocoaPods / Carthage / Prebuilt frameworks?
**CocoaPods**

Can you give us steps to reproduce with a minimal, complete, and verifiable example? Please include any specific network conditions that might be required to reproduce the problem.
N/A

Hi,

**The problem is that notification token doesn't update.**
As I noticed, to update token, sdk should call **currentEndpointProfile** method.
```
- (AWSPinpointEndpointProfile *) currentEndpointProfile {
    if (!_endpointProfile) {
        if ([_context.configuration.userDefaults objectForKey:AWSPinpointEndpointProfileKey] != nil) {
            NSData *endpointProfileData = [_context.configuration.userDefaults objectForKey:AWSPinpointEndpointProfileKey];
            AWSPinpointEndpointProfile *endpointProfile = [NSKeyedUnarchiver unarchiveObjectWithData:endpointProfileData];
            if ([endpointProfile.applicationId isEqualToString:self.context.configuration.appId]) {
                // This is to verify that same appId is being used. Anyone can modify the plist and test with a different app id
                self.endpointProfile = endpointProfile;
            } else {
                self.endpointProfile = [[AWSPinpointEndpointProfile alloc] initWithContext: self.context];
                @synchronized (self) {
                    [self.context.configuration.userDefaults removeObjectForKey:AWSPinpointEndpointProfileKey];
                    [self.context.configuration.userDefaults synchronize];
                }
            }
        } else {
            self.endpointProfile = [[AWSPinpointEndpointProfile alloc] initWithContext: self.context];
        }
    }
...
```
and **currentEndpointProfile** should initialize new **AWSPinpointEndpointProfile** object. Because initialization of **AWSPinpointEndpointProfile** updates notification device token. 
Like this:

```
...
  NSString *deviceTokenString = [[[[userDefaults objectForKey:AWSDeviceTokenKey] description] stringByTrimmingCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@"<>"]] stringByReplacingOccurrencesOfString:@" " withString:@""];
        
        _applicationId = applicationId;
        _endpointId = endpointId;
        _channelType  = debug? DEBUG_CHANNEL_TYPE : CHANNEL_TYPE;
        _address = deviceTokenString;
...
```

**BUT**

method **currentEndpointProfile** will never update device token because of condition **if (!_endpointProfile) {** that not allow create new object AWSPinpointEndpointProfile.

Thank you,
Alexander

**Comment by balaji2591** [comment]
Hi @karpov3 ,
Thanks for reporting this issue. This will be fixed in the next release.

**Comment by rohandubal** [comment]
Fixed with 2.6.16 release.
